src/OrchardCore/OrchardCore.Setup.Abstractions/TenantSetupHandlerBase.cs

```diff
@@ -0,0 +1,17 @@
+using System.Threading.Tasks;
+using OrchardCore.Setup.Events;
+using OrchardCore.Setup.Services;
+
+namespace OrchardCore.Setup;
+
+public class TenantSetupHandlerBase : ITenantSetupHandler
+{
+    public virtual Task FailedAsync(SetupContext context)
+        => Task.CompletedTask;
+
+    public virtual Task SetupAsync(SetupContext context)
+        => Task.CompletedTask;
+
+    public virtual Task SucceededAsync()
+        => Task.CompletedTask;
```

Comment:
Maybe in one line but as you want.

But in the same order than in the interface, see the related comment.